src/main/java/uk/gov/hmcts/reform/hmc/api/mapper/AutomatedHearingTransactionRequestMapper.java

```diff
@@ -0,0 +1,575 @@
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.AND;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.APPLICANT;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.C100;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.CASE_FILE_VIEW;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.CASE_SUB_TYPE;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.CASE_TYPE;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.CATEGORY_VALUE;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.COURT;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.EMPTY;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.EMPTY_STRING;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.FL401;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.ONE;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.ORGANISATION;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.PF0002;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.PF0013;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.PF0015;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.PF0018;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.PLUS_SIGN;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.RA;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.RA0042;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.RESPONDENT;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.RE_MINOR;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.SM;
+import static uk.gov.hmcts.reform.hmc.api.utils.Constants.SM0002;
+
+@Slf4j
+public final class AutomatedHearingTransactionRequestMapper {
+
+    private static boolean caseAdditionalSecurityFlag;
+
+    private AutomatedHearingTransactionRequestMapper() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static AutomatedHearingRequest mappingHearingTransactionRequest(CaseData caseData,String ccdBaseUrl) {
+
+        String publicCaseNameMapper = EMPTY;
+        if (C100.equals(caseData.getCaseTypeOfApplication())) {
+            publicCaseNameMapper = RE_MINOR;
+        } else if (FL401.equals(caseData.getCaseTypeOfApplication())) {
+            PartyDetails applicantMap = caseData.getApplicantsFL401();
+            PartyDetails respondentTableMap = caseData.getRespondentsFL401();
+            publicCaseNameMapper = (applicantMap != null && respondentTableMap != null)
+                ? applicantMap.getLastName() + AND + respondentTableMap.getLastName() : EMPTY;
+        }
+
+        List<AutomatedHearingPartyDetails> partyDetailsList = getPartyDetails(caseData);
+
+        AutomatedHearingCaseDetails caseDetail = AutomatedHearingCaseDetails.automatedHearingCaseDetailsWith()
+                .hmctsServiceCode("ABA5") //Hardcoded in prl-cos-api
```

Comment:
move this to constant

Reply:
Thank you @shashisk97 Moved to the constant ...